cds_ils/ldap/serializers.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015-2018 CERN.
+#
+# cds-migrator-kit is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""CDS-ILS ldap serializers."""
+import sys
+
+from invenio_accounts.models import User
+from invenio_oauthclient.models import UserIdentity
+from invenio_userprofiles import UserProfile
+
+
+class LDAPUser:
```

Comment:
Just in general: to me this does look more a way to force the OOP :) LDAPUser looks like more to a data structure, and not to an obj with behavior. It might be better to create a `factory` function instead.
The "contract/interface" of this object says that you can call the various methods, which instead it looks like they are only needed when creating the obj and should not be really used otherwise.

```
create_user(...):
    return LDAPUser(...)
```
and is should not be possible to create/modify the obj (immutable).

Reply:
yes, the class felt unnatural to me as well, but it was my go-to pattern. I will ask for more details on the factory